### PR TITLE
Feature/31 geolocation question type

### LINKED ISSUE #31: Geolocation Question Type
## TODO/DONE

- [x] Get the location
- [x] Map Preview
- [x] Unit tests

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Get the location
- [x] Map Preview
- [ ] Unit tests
  - [x] Library loc and expo-location
  - [x] TypeGeo
  - [ ] MapView

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204868663470092

### CODE REVIEW

**Comment by ifirmawan** [comment]
Hi mas @dedenbangkit  I'll update the unit tests in a separate PR. Thank you

File: app/src/navigation/index.js
Comment (dedenbangkit) [inline app/src/navigation/index.js:1]: Where is the test for MapViewPage component?

File: app/src/pages/MapView.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { View, StyleSheet, ActivityIndicator, Button, Platform, ToastAndroid } from 'react-native';
+import { WebView } from 'react-native-webview';
+import { Asset } from 'expo-asset';
+import * as FileSystem from 'expo-file-system';
+import { MapState } from '../store';
+import { loc } from '../lib';
+
+const MapView = ({ navigation, route }) => {
+  const [htmlContent, setHtmlContent] = React.useState(null);
```
Comment (dedenbangkit) [inline app/src/pages/MapView.js]: useState

File: app/src/pages/MapView.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import React from 'react';
```
Comment (dedenbangkit) [inline app/src/pages/MapView.js]:
```jsx
import React, { useState, useRef, useEffect } from 'react';
```

File: app/src/pages/MapView.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { WebView } from 'react-native-webview';
+import { Asset } from 'expo-asset';
+import * as FileSystem from 'expo-file-system';
+import { MapState } from '../store';
+import { loc } from '../lib';
+
+const MapView = ({ navigation, route }) => {
+  const [htmlContent, setHtmlContent] = React.useState(null);
+  const [loading, setLoading] = React.useState(true);
+  const webViewRef = React.useRef(null);
```
Comment (dedenbangkit) [inline app/src/pages/MapView.js]: useRef

File: app/src/pages/MapView.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  const loadHtml = async () => {
+    const [{ localUri }] = await Asset.loadAsync(require('../../assets/map.html'));
+    let fileContents = await FileSystem.readAsStringAsync(localUri);
+    const { latitude: lat, longitude: lng } = route?.params;
+    fileContents = fileContents.replace(/{{latitude}}/g, lat).replace(/{{longitude}}/g, lng);
+    setHtmlContent(fileContents);
+  };
+
+  React.useEffect(() => {
```
Comment (dedenbangkit) [inline app/src/pages/MapView.js]: useEffect

File: app/src/pages/MapView.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    loadHtml();
+  }, []);
+
+  React.useEffect(() => {
```
Comment (dedenbangkit) [inline app/src/pages/MapView.js]: useEffect

File: app/src/form/FormContainer.js [hunk 6]
```diff
@@ -10,7 +10,6 @@ import { transformForm } from './lib';
 // TODO:: Allow other not supported yet
 // TODO:: Repeat group not supported yet
 // TODO:: Cascade not supported yet
-// TODO:: Geo not supported yet
 
 const FormContainer = ({ forms, initialValues = {} }) => {
   const formRef = React.useRef();
```
Comment (dedenbangkit) [inline app/src/form/FormContainer.js]: useRef

File: app/src/form/components/QuestionField.js
Comment (dedenbangkit) [inline app/src/form/components/QuestionField.js:1]: QuestionField with TypeGeo test

File: app/src/form/fields/TypeGeo.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    if ((text && !values?.[id]) || (values?.[id] && values[id] !== text)) {
+      onChange(id, text);
+    }
+  }, [values, id, text]);
+
+  const [latText, lngText] = text?.split('|');
+  return (
+    <View>
+      <FieldLabel keyform={keyform} name={name} />
+      <View style={styles?.inputGeoContainer}>
```
Comment (dedenbangkit) [inline app/src/form/fields/TypeGeo.js]: why `styles?.inputGeoContainer` and not `styles.inputGeoContainer`?

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
Hi @ifirmawan please don't use `React.*` directly

File: app/src/navigation/index.js
Comment (ifirmawan) [reply app/src/navigation/index.js:1]: Actually, I'll make it in the next PR mas and I'll try to increase the coverage

Comment (ifirmawan) [reply app/src/navigation/index.js:1]: If I just make a snapshot it looks like this, 
Screenshot_20230704_211623

so it's taking me some time to sort it out

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**